File: app/src/main/java/com/example/readative/room/entity/Quote.java
package com.example.readative.room.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Quote {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "quote_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "book_owner_id")
    public long bookOwnerId;

    @NonNull
    @ColumnInfo(name = "quote_page")
    public int page;
    @NonNull
    @ColumnInfo(name = "quote_line")
    public int line;
    @NonNull
    @ColumnInfo(name = "quote_column")
    public int column;
    @NonNull
    @ColumnInfo(name = "quote_length")
    public int length;
    @NonNull
    @ColumnInfo(name = "quote_text")
    public String text;
    @NonNull
    @ColumnInfo(name = "quote_comment")
    public String comment;
    @NonNull
    @ColumnInfo(name = "quote_color")
    public int highlightColor;
}

File: app/src/main/java/com/example/readative/feature_book/data/data_source/models/Quote.kt
package com.example.readative.feature_book.data.data_source.models

import androidx.compose.ui.graphics.Color
import androidx.room.*
import androidx.room.ForeignKey.CASCADE
import com.example.readative.feature_book.data.data_source.models.Book

@Entity(
    tableName = "quotes",
    foreignKeys = [
        ForeignKey(
            entity = Book::class,
            parentColumns = ["id"],
            childColumns = ["book_id"],
            onUpdate = CASCADE,
            onDelete = CASCADE
        )
    ]
)
data class Quote(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "book_id") override val bookId: Long,
    @ColumnInfo(name = "text") val text: String,
    @ColumnInfo(name = "comment") val comment: String,
    @ColumnInfo(name = "page") val page: Int,
    @ColumnInfo(name = "start_index") val start: Int,
    @ColumnInfo(name = "end_index") val end: Int,
    @ColumnInfo(name = "highlight_color") val highlight: Color = Color(0x44000000)
) : ReadativeEntity, BookIdEntity
